### PR TITLE
Update charlievieth/fastwalk to use forward-slashes on WSL and MSYS

### PR DESCRIPTION
**Pull request by charlievieth**
This commit changes FZF to enforce that all paths are joined with forward-slashes when running on WSL (Windows Subsystem for Linux) even when the FZF binary was compiled for Windows.

Update: github.com/charlievieth/fastwalk to v1.0.5
Fixes:  https://github.com/junegunn/fzf/issues/3859

### CODE REVIEW

**Comment by charlievieth** [comment]
**NOTE:** This is just a draft and is incompatible with the `-walker-path-sep` option since (it only allows the option to force forward slashes, but does auto-detect when they're needed on WSL see: [DefaultToSlash](https://pkg.go.dev/github.com/charlievieth/fastwalk#DefaultToSlash)).

**Comment by junegunn** [comment]
Thanks, I wish we could avoid adding `--walker-path-sep`.

Defaulting to `/` on WSL sounds great. But we may also do that on MSYS2.

Related: https://github.com/sharkdp/fd/pull/730

**Comment by junegunn** [comment]
> incompatible with the `-walker-path-sep` option since

With this and #3909, we can safely remove `--walker-path-sep`.

File: src/reader.go [hunk 1]
```diff
@@ -235,7 +235,11 @@ func isSymlinkToDir(path string, de os.DirEntry) bool {
 
 func (r *Reader) readFiles(root string, opts walkerOpts, ignores []string) bool {
 	r.killed = false
-	conf := fastwalk.Config{Follow: opts.follow}
+	conf := fastwalk.Config{
+		Follow: opts.follow,
+		// Use forward slashes when running a Windows binary under WSL.
```
Comment (junegunn) [inline src/reader.go:240]:
```suggestion
		// Use forward slashes when running a Windows binary under WSL or MSYS
```

**Comment by junegunn** [comment]
So I removed `--walker-path-sep`, updated the PR branch, applied the latest version of fastwalk (`go get github.com/charlievieth/fastwalk@master`), tested it on MSYS2, but it wasn't working as expected.

Turned out that `filepath.Clean(path)` is replacing `/` back to `\` 😢 

https://github.com/charlievieth/fzf/blob/fe9dc4c75617007ae15c205db8516fca212a6e30/src/reader.go#L247

**Comment by junegunn** [comment]
We probably don't need the whole `filepath.Clean(path)`. We should be fine, even if we just remove the `./` or `.\`.

**Comment by junegunn** [comment]
9b30ec559c908b83142b8f6d5ffad5524df427b3 removes `filepath.Clean`. It will not prefectly handle `fzf --walker-root ./././../././`, but I don't care.

**Pull request merged**